improve python quality

alexandrupaler opened the pull request:
solves #191

95-martin-orion commented on python/cirq_interface/fsim_gate.py:
```diff
@@ -25,9 +25,10 @@
 import math
 from typing import Optional
 
+import cirq
```
I think the warning on this line is because we also have `from cirq ...` further down. We should remove the `from cirq` imports and update their call sites to use the full `cirq.{packages}...` path.

alexandrupaler replied:
I keep it like this and in the next PR move to 0.6 Cirq

alexandrupaler commented:
should solve #174 and #177 . This PR also prepares the way to remove temp files. Added data_storage_interface.py

95-martin-orion commented on python/cirq_interface/data_storage_interface.py:
```diff
@@ -0,0 +1,35 @@
+
+        try:
+            # https://stackoverflow.com/questions/36069102/pytest-fixture-finalizer-typeerror-nonetype-object-is-not-callable
+            yield
```
Could you clarify how this works? My reading of the linked page (plus the ["docs" link](https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code) on that page) is that this `yield` --> `os.close()` structure would need to be contained in a `pytest.fixture` method to behave as expected. (More info on how to use pytest.fixture is also given at that link)

alexandrupaler replied:
Short explanation: pure luck that it works. Elaborated (potential explanation): The yield transforms the method into a generator, and this fact solves a race condition in the destruction of the object.

95-martin-orion replied:
As far as I can tell, the link in the comment is specific to `pytest.fixture`, and isn't related to this code - searching for "python yield destructor" turns up zero relevant results. Even if this appears to resolve the issue, we shouldn't rely on it.

alexandrupaler replied:
The link in the comment mentions the yield as a solution that has applicability to the fixture.

95-martin-orion replied:
Discussed offline - solution for now is to omit the `yield`, leaving #174 and #177 unresolved for now. Those can be resolved in a future PR.

95-martin-orion commented on python/cirq_interface/data_storage_interface.py:
```diff
@@ -0,0 +1,35 @@
+import tempfile
+import os
+
+class DataStorageInterface:
```
Just a question, no change required: is the intent to later modify this class to support both temp-file and non-temp-file behavior?

alexandrupaler replied:
Yes

95-martin-orion commented on tests/python/simulation_test.py:
```diff
@@ -596,13 +594,13 @@ def test_simulation_with_fsim_gates(x):
     }
 
-    print(options)
+    # print(options)
```
remove comment?

95-martin-orion commented on tests/python/qflex_circuit_test.py:
```diff
@@ -13,6 +13,42 @@
+                             dummy_order,
+                             allow_decomposition=True)
+
+    # should be ok
```
This comment can be removed, as it's implied by `assert`. Same below.

alexandrupaler commented:
174 and 177 are solved only because of the yield in the destructor. Please keep the PR like this, if it is not too much to ask, because there is already another PR to remove temp files by Salvatore, and we would duplicate/overlap too much for simple things.

alexandrupaler commented:
I am not in favour of removing files altogether, but I think I am not the one to decide. However, the version of the code three weeks ago had no temp files, then temp files appeared and introduced very strange behaviours which took days to debug and understand, and I had to write my code around them. Now I find actually temp files useful for debugging, and that is the reason I introduced the data interface.
